### DISCUSSION TITLE
TSS Gatsby example project

### DISCUSSION THREAD

**Discussion by Deckstar** [garronej/tss-react#67] [Show and tell]
Hey! 🙂 Thanks a lot for giving my article a shoutout on your docs page!

I wanted to ask — would it be okay if we replaced the example project? I've created a [new simplified project](https://github.com/Deckstar/gatsby-tss-example) that should be much better suited as a showcase 😃 (Talking about [this page](https://docs.tss-react.dev/ssr/gatsby) in the docs).

I realise that you're in the process of making a Gatsby plugin, but would it be alright to still change the example in the mean time? Having my personal page be used as an example for all to see just feels like a little bit more attention than I'm comfortable with at the moment. I'm actually a bit of a shy guy 😄

**Comment by garronej** [comment]
Hi @Deckstar,  

Sure, sorry, I should have asked you before using your portfolio website as an example.  
Thank you for putting together an other reference project.   

It's updated.

**Reply by Deckstar** [reply]
Hahah no worries. I mean it is a public repository so technically anyone can still see it. It just felt a bit wrong to me to use it as an example 😄 

But yeah, no harm done! Thanks for the free advertising? 😄

**Comment by garronej** [comment]
Wow. I just had a look at https://github.com/Deckstar/gatsby-tss-example.  
It's really great thank you so much.  
I don't even know if a gatsby plugin is necessary still.

**Reply by Deckstar** [reply]
Thanks a lot for your kind words! Glad I could help! 😃 I like tss-react a lot so I'm glad to help out!

I think I would skip the plug-in as well if I were you, to be honest. The MUI plug-in felt a bit fragile to me because it had to kind of "hack" the file system and bundler so it could import the MUI cache from a custom import route. So it didn't have any intelligence for me, which isn't very TS-friendly. Though I guess it's not a big deal since the actual user would never see this hack 😄

But the bigger issue is that tss-react is supposed to be MUI-independent but I'm sure a lot of people (including myself) would like to use it with MUI, so presumably both libraries would have to be handled somehow in the plug-in, kind of like how I did it in the example. There might be a clever way to have multiple SSR plugins at the same time but I couldn't find it last time.

So imo for users it may just be much easier (or at least more reliable) to manipulate the Gatsby config files directly, instead of downloading a plug-in that might not work well with other plug-ins.
